### PR TITLE
Custom string for TextOverflow.ellipsis #26748

### PR DESCRIPTION
**Pull request by vinaykumarrock986612**
Screenshot_1699201404

Screenshot_1699201424

*Addressed #26748 by adding a high level attribute to assign a custom String to TextOverflow.ellipsis*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by goderbauer** [comment]
Looks like this implementation is currently failing some checks and it is also missing tests.

**Comment by goderbauer** [comment]
@vinaykumarrock986612 There is another PR in flight right now with some more discussion that is trying to implement the same thing. Please take a look at https://github.com/flutter/flutter/pull/104337 and the discussion there. The work appears to have stalled there. Maybe you're interested in picking up where that PR was left of?

FYI @LongCatIsLooong

File: packages/flutter/lib/src/rendering/paragraph.dart [hunk 1]
```diff
@@ -503,6 +504,15 @@ class RenderParagraph extends RenderBox with ContainerRenderObjectMixin<RenderBo
     markNeedsLayout();
   }
 
+  /// TextOverflow.ellipsis's value.
+  String? get ellipsis => _textPainter.ellipsis;
+  set ellipsis(String? value) {
+    if(value != null){
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/rendering/paragraph.dart:510]: This doesn't look correct?

File: packages/flutter/lib/src/rendering/paragraph.dart [hunk 2]
```diff
@@ -503,6 +504,15 @@ class RenderParagraph extends RenderBox with ContainerRenderObjectMixin<RenderBo
...
+      _textPainter.ellipsis = value;
+      markNeedsLayout();
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/rendering/paragraph.dart:512]: Is the `markNeedsLayout` needed, or do we only need to repaint?

File: packages/flutter/lib/src/rendering/paragraph.dart [hunk 3]
```diff
@@ -503,6 +504,15 @@ class RenderParagraph extends RenderBox with ContainerRenderObjectMixin<RenderBo
     markNeedsLayout();
   }
 
+  /// TextOverflow.ellipsis's value.
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/rendering/paragraph.dart:507]: Maybe: The string to replace the omitted text, when `textOverflow` is set to `TextOverflow....` and the text overflows.

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/rendering/paragraph.dart:507]: also consider describing the default value, as well as edge cases such as what happens if the ellipsis string itself overflows.

File: packages/flutter/lib/src/widgets/basic.dart [hunk 4]
```diff
@@ -5815,6 +5816,10 @@ class RichText extends MultiChildRenderObjectWidget {
   /// How visual overflow should be handled.
   final TextOverflow overflow;
 
+  /// By default the TextOverflow.ellipsis shows "..."
+  /// Use [ellipsisValue] to show custom String
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/widgets/basic.dart:5820]: See the comment above regarding documentation. You could use a dartdoc macro to reuse a piece of documentation (search for `@macro` and `@template` for examples).

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/basic.dart:5820]: Same for the API documentation in text.dart.

**Comment by LongCatIsLooong** [comment]
also: 

> ║ /b/s/w/ir/x/w/Á flutter sdk/packages/flutter/lib/src/rendering/paragraph.dart:510: no space after flow control statement

**Comment by goderbauer** [comment]
(triage) @vinaykumarrock986612 Do you still have plans to address the feedback given above?

**Comment by vinaykumarrock986612** [comment]
> (triage) @vinaykumarrock986612 Do you still have plans to address the feedback given above?

@goderbauer I'll surely work on it in free time.

**Comment by goderbauer** [comment]
(triage) I am going to close this one since it is getting stale. Feel free to reopen this when you have the time to work an this gain.

**Pull request closed without merging**